domain/consensus/timelock_CSV_test.go

```diff
@@ -0,0 +1,524 @@
+		if err != nil {
+			t.Fatalf("Error creating blockA: %v", err)
+		}
+		blockBHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockAHash}, nil, nil)
+		if err != nil {
+			t.Fatalf("Error creating blockB: %v", err)
+		}
+		blockCHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockBHash}, nil, nil)
+		if err != nil {
+			t.Fatalf("Error creating blockC: %v", err)
+		}
+		blockDHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockCHash}, nil, nil)
+		if err != nil {
+			t.Fatalf("Error creating blockD: %v", err)
+		}
+		blockD, err := testConsensus.GetBlock(blockDHash)
+		if err != nil {
+			t.Fatalf("Failed getting blockD: %v", err)
+		}
+		fees := uint64(1)
+		// Create a CSV script
+		numOfDAAScoreToWait := uint64(10)
+		redeemScriptCSV, err := createScriptCSV(numOfDAAScoreToWait)
+		if err != nil {
+			t.Fatalf("Failed to create a script using createCheckSequenceVerifyScript: %v", err)
+		}
+		p2shScriptCSV, err := txscript.PayToScriptHashScript(redeemScriptCSV)
+		if err != nil {
+			t.Fatalf("Failed to create a pay-to-script-hash script : %v", err)
+		}
+		scriptPublicKeyCSV := externalapi.ScriptPublicKey{
+			Version: constants.MaxScriptPublicKeyVersion,
+			Script:  p2shScriptCSV,
+		}
+		transactionWithLockedOutput, err := createTransactionWithLockedOutput(
+			blockD.Transactions[transactionhelper.CoinbaseTransactionIndex], fees, &scriptPublicKeyCSV)
+		if err != nil {
+			t.Fatalf("Error in createTransactionWithLockedOutput: %v", err)
+		}
+		// BlockE contains the locked output (locked by CSV).
+		// This block should be valid since CSV script locked only the output.
+		blockEHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockDHash}, nil,
+			[]*externalapi.DomainTransaction{transactionWithLockedOutput})
+		if err != nil {
+			t.Fatalf("Error creating blockE: %v", err)
+		}
+		// bit 62 of sequence defines if it's conditioned by DAA score(set to 0) or by time (set to 1).
+		sequenceTypeFlag := 0
+		// Create a transaction that tries to spend the locked output.
+		transactionThatSpentTheLockedOutput, err := createTransactionThatSpentTheLockedOutputRelativeLock(transactionWithLockedOutput,
+			fees, redeemScriptCSV, numOfDAAScoreToWait, sequenceTypeFlag, blockEHash, &testConsensus)
+		if err != nil {
+			t.Fatalf("Error creating transactionThatSpentTheLockedOutput: %v", err)
+		}
+		// Add a block that contains a transaction that spends the locked output before the time, and therefore should be failed.
+		_, _, err = testConsensus.AddBlock([]*externalapi.DomainHash{blockEHash}, nil,
+			[]*externalapi.DomainTransaction{transactionThatSpentTheLockedOutput})
+		if err == nil || !errors.Is(err, ruleerrors.ErrUnfinalizedTx) {
+			t.Fatalf("Expected block to be invalid with err: %v, instead found: %v", ruleerrors.ErrUnfinalizedTx, err)
+		}
+		// Adds blocks until it reaches the DAA score target, so the locked output will be released.
```

Comment:
This comment should be above the loop, where here you should write something along the lines of "Add block that accepts the locked output"